torch/onnx/_internal/exporter/_fx_passes.py

```diff
@@ -45,3 +45,49 @@ def remove_assertion_nodes(graph_module: torch.fx.GraphModule) -> torch.fx.Graph
             graph_module.graph.erase_node(node)
     graph_module.recompile()
     return graph_module
+
+
+def insert_flatten_between_transpose_and_view(
+        exported_program: torch.export.ExportedProgram
+) -> torch.export.ExportedProgram:
+    """Modifies the module inplace to insert a node 'flatten' between a node 'transpose' followed by a node 'view'.
+
+    The modification takes place inplace.
+
+    Remove after issue https://github.com/pytorch/pytorch/issues/136543 is fixed.
+    """
+    modified = False
+    graph = exported_program.graph_module.graph
+    for node in graph.nodes:
+        if (node.op != "call_method" or node.target != "transpose") and (
+            node.op != "call_function"
+            or not hasattr(node.target, "name")
+            or node.target.name() != "aten::transpose.int"
+        ):
+            continue
+        insert = False
+        for user in node.users:
+            if (user.op == "call_method" and user.target == "view") or (
+                user.op == "call_function"
+                and hasattr(node.target, "name")
+                and user.target.name() == "aten::view"
+            ):
+                insert = True
+                break
+        if not insert:
+            continue
+
+        modified = True
+        with graph.inserting_after(node):
+            new_node = graph.call_method("flatten", args=(node,))
```

Comment:
Do we need to call the actual function? `torch.ops.aten.contiguous`? Maybe call_method will translate this for us or?

Reply:
Do we need call_function instead?

Reply:
It works so I assume the decompositions takes care of it.

Reply:
It does work because the mechanism doesn’t care, but I think we should still maintain the exported program assumptions linked above (aka no call methods)

Reply:
I'll remove then. When I started to implement, I saw call_method but I don't know why I don't see them now.